plugins/services/src/js/reducers/serviceForm/Volumes.js

```diff
@@ -167,7 +186,9 @@ function reduceVolumes(state, {type, path, value}) {
     this.localVolumes
       .filter(filterHostVolumes.bind(this))
       .map(mapLocalVolumes),
-    this.externalVolumes);
+    this.externalVolumes.map(
+      mapExternalVolumes.bind(this)
```

Comment:
Wonder if it makes sense to only pass the needed properties instead of the entire internal state...